recipe/gen_patch_json.py

```diff
@@ -808,6 +808,27 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         if i >= 0:
             deps[i] = "cudatoolkit >=11.2,<12.0a0"
 
+        # cuda-python 11.7.1 is ABI-breaking, currently it impacts rmm & numba
+        if record_name == "rmm":
+            if pkg_resources.parse_version(record["version"]) < pkg_resources.parse_version("22.08"):
+                # these were built with cuda-python <= 11.7.0
+                _pin_stricter(fn, record, "cuda-python", max_pin="x", upper_bound="11.7.1")
+            if (pkg_resources.parse_version(record["version"]) == pkg_resources.parse_version("22.08")
+                and record['build_number'] == 0):
+                # this is built with cuda-python 11.7.1
+                _replace_pin("cuda-python >=11.7,<12.0a0", "cuda-python >=11.7.1,<12.0a0", deps, record)
+        if (record_name == "numba"
+            and pkg_resources.parse_version(record["version"]) <= pkg_resources.parse_version("0.56.2")):
+            for i, dep in enumerate(record.get("constrains", [])):
+                if dep.startswith("cuda-python"):
+                    record["constrains"][i] = "cuda-python >=11.6,<11.7.1"
+                    break
+        if (record_name == "cuda-python"
+            and pkg_resources.parse_version(record["version"]) == pkg_resources.parse_version("11.7.1")
+            and record['build_number'] == 0):
+            # add run_exports
+            deps.append("cuda-python >=11.7.1,<12.0a0")
```

Comment:
Q: Does this achieve this desired effect?
```yaml
run_exports:
  - {{ pin_subpackage('cuda-python') }}
```

Reply:
No, you cannot patch `run_exports`. You have to fix it in the recipe for the old version (11.7.0) and the new version (11.7.1)

Reply:
Thanks, @isuruf. We're fixing the new version now (https://github.com/conda-forge/cuda-python-feedstock/pull/17), but I don't think we can afford to rebuild all prior versions?

Reply:
We could patch packages that depend on `cuda-python` to adjust their version requirement

Reply:
@jakirkham that's exactly what I did with this PR.

Reply:
You need to rebuild at least 11.7.0 with the correct run_exports. Otherwise this patch will affect packages built in the future, which we dont want.

Reply:
@isuruf I do not understand how future packages (`rmm`? `numba`? `cuda-python`?) could be impacted, could you elaborate? Thanks.